core/consensus/aura/src/lib.rs

```diff
@@ -112,32 +112,61 @@ fn inherent_to_common_error(err: RuntimeString) -> consensus_common::Error {
 }
 
 /// A digest item which is usable with aura consensus.
-pub trait CompatibleDigestItem<T: Pair>: Sized {
+pub trait CompatibleDigestItem<U, V>: Sized {
 	/// Construct a digest item which is a slot number and a signature on the
 	/// hash.
-	fn aura_seal(slot_number: u64, signature: Signature<T>) -> Self;
+	fn aura_seal(slot_number: U, signature: V) -> Self;
 
 	/// If this item is an Aura seal, return the slot number and signature.
-	fn as_aura_seal(&self) -> Option<(u64, Signature<T>)>;
+	fn as_aura_seal(&self) -> Option<(U, V)>;
 }
 
-impl<T: Pair, Hash, AuthorityId> CompatibleDigestItem<T> for generic::DigestItem<Hash, AuthorityId, Signature<T>>
-	where T::Signature: Clone
+#[deprecated]
+pub type DeprecatedDigestItem<T> = CompatibleDigestItem<u64, T>;
+
+pub type ModernDigestItem = CompatibleDigestItem<ConsensusEngineId, Vec<u8>>;
+
+pub trait GoodDigestItem<T: Pair>: CompatibleDigestItem<u64, T::Signature> + CompatibleDigestItem<ConsensusEngineId, Vec<u8>> {}
+
+#[deprecated]
+impl<Pub, Sig, Hash> CompatibleDigestItem<u64, Sig> for generic::DigestItem<Hash, Pub, Sig>
+	where Sig: Clone,
 {
 	/// Construct a digest item which is a slot number and a signature on the
 	/// hash.
-	fn aura_seal(slot_number: u64, signature: Signature<T>) -> Self {
+	fn aura_seal(slot_number: u64, signature: Sig) -> Self {
 		generic::DigestItem::Seal(slot_number, signature)
 	}
 	/// If this item is an Aura seal, return the slot number and signature.
-	fn as_aura_seal(&self) -> Option<(u64, Signature<T>)> {
+	fn as_aura_seal(&self) -> Option<(u64, Sig)> {
 		match self {
 			generic::DigestItem::Seal(slot, ref sig) => Some((*slot, (*sig).clone())),
 			_ => None
 		}
 	}
 }
 
+impl<Pub, Sig, Hash> CompatibleDigestItem<ConsensusEngineId, Vec<u8>> for generic::DigestItem<Hash, Pub, Sig>
+	where Sig: Clone,
+{
+	/// Construct a digest item which is a slot number and a signature on the
+	/// hash.
+	fn aura_seal(engine_id: ConsensusEngineId, seal: Vec<u8>) -> Self {
```

Comment:
Isn't "aura" the consensus engine? why does an `aura_seal` require/allow to provide a `ConsensusEngineId`? Shouldn't it always be `[b"a", b"u", b"r", b"a"]`?

Reply:
Should I replace this argument by `()`?  Or should I use a tuple for the legacy version?

Reply:
I'd say the tuple for the legacy makes more sense.

Reply:
I didn’t implement the type aliases, but I did the rest.